com.unity.netcode.gameobjects/Editor/NetworkBehaviourEditor.cs

```diff
@@ -211,5 +211,79 @@ public override void OnInspectorGUI()
+        /// </summary>
+        private void OnEnable()
+        {
+            // When we first add a NetworkBehaviour this editor will be enabled
+            // so we go ahead and check for an already existing NetworkObject here
+            CheckForNetworkObject((target as NetworkBehaviour).gameObject);
+        }
+
+        internal const string AutoAddNetworkObjectIfNoneExists = "AutoAdd-NetworkObject-When-None-Exist";
+
+        /// <summary>
+        /// Used to determine if a GameObject has one or more NetworkBehaviours but
+        /// does not already have a NetworkObject component.  If not it will notify
+        /// the user that NetworkBehaviours require a NetworkObject.
+        /// </summary>
+        public static void CheckForNetworkObject(GameObject gameObject, bool networkObjectRemoved = false)
+        {
+            // If there are no NetworkBehaviours or no gameObject, then exit early
+            if (gameObject == null || gameObject.GetComponent<NetworkBehaviour>() == null && gameObject.GetComponentInChildren<NetworkBehaviour>() == null)
+            {
+                return;
+            }
+
+            // Otherwise, check to see if there is a NetworkObject and if not notify the user that NetworkBehaviours
+            // require that the relative GameObject has a NetworkObject component.
+            var networkObject = gameObject.GetComponent<NetworkObject>();
+            if (networkObject == null)
+            {
+                networkObject = gameObject.GetComponentInChildren<NetworkObject>();
+
+                if (networkObject == null)
+                {
+                    // If we are removing a NetworkObject but there is still one or more NetworkBehaviour components
+                    // and the user has already turned "Auto-Add NetworkObject" on when first notified about the requirement
+                    // then just send a reminder to the user why the NetworkObject they just deleted seemingly "re-appeared"
+                    // again.
+                    if (networkObjectRemoved && EditorPrefs.HasKey(AutoAddNetworkObjectIfNoneExists) && EditorPrefs.GetBool(AutoAddNetworkObjectIfNoneExists))
+                    {
+                        Debug.LogWarning($"{gameObject.name} still has {nameof(NetworkBehaviour)}s and Auto-Add NetworkObjects is enabled. A NetworkObject is being added back to {gameObject.name}.");
+                        Debug.Log($"To reset Auto-Add NetworkObjects: Select the Netcode->General->Reset Auto-Add NetworkObject menu item.");
+                    }
+
+                    // Notify and provide the option to add it one time, always add a NetworkObject, or do nothing and let the user manually add it
+                    if (EditorUtility.DisplayDialog($"{nameof(NetworkBehaviour)}s require a {nameof(NetworkObject)}",
+                    $"{gameObject.name} does not have a {nameof(NetworkObject)} component.  Would you like to add one now?", "Yes", "No (manually add it)",
```

Comment:
I really like that this popup gives the option to automatically add it. I was actually coming into this expecting to bring up a concern about being annoyed by popups if I add the objects in the wrong order, and here you already thought of that and made a great solution. I didn't even have a solution in mind, I was just going to ask "do we care?" - great work!

Reply:
Though as a minor thing, the text of this popup disagrees with the comment... the comment suggests this dialogue will give the user the choice between "once", "always", or "never", but this text gives the choice between "yes", or "no".

Reply:
(Though I may be misreading that, given I don't know what `DialogOptOutDecisionType.ForThisMachine` does.)

Reply:
Yeah... this is a "Unity Thing" that basically gives the option to always/automatically select "Yes" for the scenario that invokes that dialog box (i.e. so a user can see it once, select the extended "Yes-Do not show this...", and then it will always automatically add a NetworkObject component if you first add a NetworkBehaviour component to a GameObject with no NetworkObject component.

Reply:
Ah, neat. Comment withdrawn then!